src/components/partner/partner.service.ts

```diff
@@ -189,9 +189,48 @@ export class PartnerService {
     { filter, ...input }: PartnerListInput,
     session: Session
   ): Promise<PartnerListOutput> {
-    const query = this.repo.list({ filter, ...input }, session);
+    if (await this.authorizationService.canList(Partner, session)) {
+      const query = this.repo.list({ filter, ...input }, session);
+      return await runListQuery(query, input, (id) =>
+        this.readOne(id, session)
+      );
+    } else {
+      const listPartnerIds = [] as ID[];
+      const result = await this.repo
+        .listPartnersOfAllUserProjects(session, { filter, ...input })
+        .run();
```

Comment:
I don't think this direction is a good idea. It's an enormous amount of work to put onto the API every time a partner list is requested. Most of the reason to paginate is to limit the load on the API. In summary we are pulling the entire list only to filter it in app code and go back to the DB sort & paginate, and worse the do another lookup per item. That's O(n+2).

I think the logic here is if you are a member of this project with X sensitivity and Y roles then you can pass. I believe the auth service could compile that into a simple structure, which the DB query could then convert into a WHERE clause. This would remove the back and forth from the DB and still leaves us in a spot where we can merge the readOnes into the query as well bringing the number of queries to O(1).